### ISSUE TITLE
How to train the model with custom datasets 

### ISSUE THREAD

**Issue by Elmiar0642** [ELEKTRONN/elektronn3#50]
I wish to use ADNI 3D MRI dataset. I saw training script. Idk how to modify it for custom datasets, especially like ADNI.

I just wish to normally create a model object and then train the model with the 3D images I have inside a directory. I just need to visualise the feature maps at each unit.

**Comment by mdraw** [comment]
If you want to use another dataset there are in principle two ways to proceed:
- Adapt the dataset to elektronn3's expected format for data loading (usually HDF5 files of the same layout as the ones you can find in the [example data](https://github.com/ELEKTRONN/elektronn.github.io/releases/download/neuro_data_cdhw/neuro_data_cdhw.zip
)), so you can replace the data set in the `train_unet_neurodata.py` example with it.
- Or use/write custom data loading code for your dataset. It just should be a subclass of `torch.utils.data.Dataset` and return a dict that contains an `'inp'` key for input data as `np.float32` and a `'target'` key for target (label) data as `np.int64` arrays if you want to train for classification. Everything else can be freely chosen. Then you can replace the reference to `PatchCreator` in the example script by your own `Dataset` class.
